### PR TITLE
Add arc-2 quickstart guide

### PR DESCRIPTION
**Pull request by Link-**
Fixes https://github.com/github/c2c-actions-runtime/issues/2253

Following the [announcement of the new autoscaling mode](https://github.com/actions/actions-runner-controller/discussions/2174), this is a quick start guide describing the setup process.

- [Rendered version](https://github.com/actions/actions-runner-controller/blob/32355577751c81a2fe2d2d2bfb5c1a1480ebff5e/docs/preview/actions-runner-controller-2/README.md)

### CODE REVIEW

**Comment by rentziass** [comment]
Running through it on a fresh system 👀

File: docs/preview/actions-runner-controller-2/README.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    ```bash
+    NAMESPACE="arc-systems"
+    helm install arc \
+        --namespace "${NAMESPACE}" \
+        --create-namespace \
+        oci://ghcr.io/actions/actions-runner-controller-charts/actions-runner-controller-2 \
+        --version 0.1.0
+    ```
+
+1. Generate a Personal Access Token (PAT) or create and install a GitHub App. See [Creating a personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) and [Creating a GitHub App](https://docs.github.com/en/developers/apps/creating-a-github-app).
```
Comment (rentziass) [inline docs/preview/actions-runner-controller-2/README.md:44]: Can we list what rights does the PAT need for enterprise/org/repo scenarios?

File: docs/preview/actions-runner-controller-2/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+1. You're ready to install the autoscaling runner set. For additional configuration options, see [values.yaml](https://github.com/actions/actions-runner-controller/blob/master/charts/auto-scaling-runner-set/values.yaml)
+
+    ```bash
```
Comment (rentziass) [inline docs/preview/actions-runner-controller-2/README.md:50]: A few things on this section:
- Can we split this in two blocks for easier copy paste? 😬 
- there's an extra `--set` on L55
- is it worth parameterising config url and pat/app details like we did for `NAMESPACE`? Makes it for cleaner, easier to mess around with commands especially if (ask me how I know) you get the config url wrong, much easier to edit a variable then a multi-line command
- the config url is a bit confusing, maybe worth specifying how to get one (i.e. from the new runner page)?
- should we point out that `arc-runner-set` is what will be used as the name of scale set? Helm newbie here had to figure that one out 🤦

File: docs/preview/actions-runner-controller-2/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    $ kubectl get pods -n "${NAMESPACE}"
+
+    NAME                                              READY   STATUS    RESTARTS   AGE
+    arc-actions-runner-controller-2-8c74b6f95-gr7zr   1/1     Running   0          20m
+    arc-runner-set-6cd58d58-listener                  1/1     Running   0          21s
+    ```
+
+1. In your repository, create a simple test workflow as follows. The `runs-on` value should match the helm installation name you used in the previous step.
```
Comment (rentziass) [inline docs/preview/actions-runner-controller-2/README.md]: Should we specify that the repo has to have access to the scale set or do we assume we're doing everything at repo level for this guide?

**Review by rentziass — COMMENTED** [review]
Even for a complete noob like me this was really easy to follow! 👏 I've added a few comments to maybe clarify a few things.

File: docs/preview/actions-runner-controller-2/README.md (hunk 1 above)
Comment (Link-) [reply docs/preview/actions-runner-controller-2/README.md:44]: Yep, good call

Comment (Link-) [reply docs/preview/actions-runner-controller-2/README.md:44]: We have those documented here: https://github.com/actions/actions-runner-controller/blob/32355577751c81a2fe2d2d2bfb5c1a1480ebff5e/docs/authenticating-to-the-github-api.md#L1 so I'm just going to reference the existing docs

File: docs/preview/actions-runner-controller-2/README.md (hunk 2 above)
Comment (Link-) [reply docs/preview/actions-runner-controller-2/README.md:50]: Great points! 

I already mentioned below that:
> The `runs-on` value should match the helm installation name you used in the previous step.

Probably here is a better spot

File: docs/preview/actions-runner-controller-2/README.md (hunk 3 above)
Comment (Link-) [reply docs/preview/actions-runner-controller-2/README.md]: hmm it should be implied at this point, I'm not sure it's worth the extra hand-holding here

File: docs/preview/actions-runner-controller-2/README.md (hunk 2 above)
Comment (Link-) [reply docs/preview/actions-runner-controller-2/README.md:50]: I addressed these comments, can you have another check please?

**Review by rentziass — APPROVED** [review]

File: docs/preview/actions-runner-controller-2/README.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+1. Generate a Personal Access Token (PAT) or create and install a GitHub App. See [Creating a personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) and [Creating a GitHub App](https://docs.github.com/en/developers/apps/creating-a-github-app).
+    - For the list of required permissions, see [Authenticating to the GitHub API](https://github.com/actions/actions-runner-controller/blob/master/docs/authenticating-to-the-github-api.md#authenticating-to-the-github-api).
+
+1. You're ready to install the autoscaling runner set. For additional configuration options, see [values.yaml](https://github.com/actions/actions-runner-controller/blob/master/charts/auto-scaling-runner-set/values.yaml)
+    - **Choose your installation name carefully**, you will use it as the value of `runs-on` in your workflow.
+
+    ```bash
+    # Using a Personal Access Token (PAT)
+    INSTALLATION_NAME="arc-runner-set" 
+    NAMESPACE="arc-systems"
```
Comment (TingluoHuang) [inline docs/preview/actions-runner-controller-2/README.md:53]: we might want to suggest customers deploy the autoscalingrunnerset into a different namespace to show best security practices.

File: docs/preview/actions-runner-controller-2/README.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        --create-namespace \
+        --set githubConfigUrl="${GITHUB_CONFIG_URL}" \
+        --set githubConfigSecret.github_token="${GITHUB_PAT}" \
+        oci://ghcr.io/actions/actions-runner-controller-charts/auto-scaling-runner-set --version 0.1.0
+    ```
+
+    ```bash
+    # Using a GitHub App
+    INSTALLATION_NAME="arc-runner-set" 
+    NAMESPACE="arc-systems"
```
Comment (TingluoHuang) [inline docs/preview/actions-runner-controller-2/README.md:67]: same here

**Pull request merged**